shared-bindings/_pixelbuf/PixelBuf.c

```diff
@@ -87,23 +83,61 @@ STATIC mp_obj_t pixelbuf_pixelbuf_make_new(const mp_obj_type_t *type, size_t n_a
         { MP_QSTR_brightness, MP_ARG_OBJ, { .u_obj = mp_const_none } },
         { MP_QSTR_rawbuf, MP_ARG_OBJ, { .u_obj = mp_const_none } },
         { MP_QSTR_offset, MP_ARG_INT, { .u_int = 0 } },
-        { MP_QSTR_dotstar, MP_ARG_BOOL, { .u_bool = false } },
         { MP_QSTR_auto_write, MP_ARG_BOOL, {.u_bool = false} },
         { MP_QSTR_write_function, MP_ARG_OBJ, {.u_obj = mp_const_none} },
         { MP_QSTR_write_args, MP_ARG_OBJ, {.u_obj = mp_const_none} },
     };
     mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+    const char *byteorder = NULL;
+    pixelbuf_byteorder_details_t byteorder_details;
+    size_t bo_len;
+
+    if (!MP_OBJ_IS_STR(args[ARG_byteorder].u_obj))
+        mp_raise_TypeError(translate("byteorder is not a string"));
+
+    if (args[ARG_byteorder].u_obj == NULL)
+        byteorder = "BGR";
+    else 
+        byteorder = mp_obj_str_get_data(args[ARG_byteorder].u_obj, &bo_len);
+
+    if (bo_len < 3 || bo_len > 4)
+        mp_raise_ValueError(translate("Invalid byteorder string"));
+    strncpy(byteorder_details.order, byteorder, sizeof(byteorder_details.order));
+
+    byteorder_details.bpp = bo_len;
+    char *dotstar = strchr(byteorder, 'D');
+    char *r = strchr(byteorder, 'R');
+    char *g = strchr(byteorder, 'G');
+    char *b = strchr(byteorder, 'B');
+    char *w = strchr(byteorder, 'W');
+    int num_chars = (dotstar ? 1 : 0) + (w ? 1 : 0) + (r ? 1 : 0) + (g ? 1 : 0) + (b ? 1 : 0);
+    if ((num_chars < byteorder_details.bpp) || !(r && b && g))
+        mp_raise_ValueError(translate("Invalid byteorder string"));
+    byteorder_details.is_dotstar = dotstar ? true : false;
+    byteorder_details.has_white = w ? true : false;
+    byteorder_details.byteorder.r = r - byteorder;
+    byteorder_details.byteorder.g = g - byteorder;
+    byteorder_details.byteorder.b = b - byteorder;
+    byteorder_details.byteorder.w = w ? w - byteorder : 0;
+    // The dotstar brightness byte is always first (as it goes with the pixel start bits)
```

Comment:
I'd suggest having D or `P` for pulse at the start so all characters represent a byte in order. It's weird to have a suffix shift everything.

Reply:
I originally had them be D prefixes (but P could work), but it just looked wrong.  I'm happy to make that change though, as it simplifies the code.